media/js/firefox/features-sync.js

```diff
@@ -1,104 +1,126 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
-* License, v. 2.0. If a copy of the MPL was not distributed with this
-* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-(function($, Mozilla) {
+// Create namespace
+if (typeof Mozilla === 'undefined') {
+    var Mozilla = {};
+}
+
+(function(dataLayer) {
     'use strict';
 
-    window.dataLayer = window.dataLayer || [];
-
-    /* This shows five different content variations, depending on the browser/state
-     * 1. Firefox 31+ (signed-in to Sync) <-- default
-     * 2. Firefox 31+ (signed-out of Sync)
-     * 3. Firefox 30 or older
-     * 4. Firefox for Android
-     * 5. Not Firefox (any other browser)
-     */
-
-    // Variation #1: Firefox 31+ signed-in to Sync
-    // Default (do nothing)
-
-    var params = new window._SearchParams();
-    var client = window.Mozilla.Client;
-    var fxMasterVersion = client.FirefoxMajorVersion;
-    var state = 'Unknown';
-    var syncCapable = false;
-    var body = $('body');
-
-    var swapState = function(stateClass) {
-        body.removeClass('state-default');
-        body.addClass(stateClass);
+    var SyncPage = {
+        body: document.getElementsByTagName('body')[0],
+        dataLayer: dataLayer,
+        state: 'Unknown',
+        syncCapable: false
     };
 
-    // Variations 1-4 are Firefox
-    if (client.isFirefox) {
-
-        // Variation #4: Firefox for Android
-        if (client.isFirefoxAndroid) {
-            swapState('state-fx-android');
-            state = 'Firefox for Android';
-
-        // Variation #1-3: Firefox for Desktop
-        } else if (client.isFirefoxDesktop) {
-
-            if (fxMasterVersion >= 31) {
-
-                // Set syncCapable so we know not to send tracking info
-                // again later
-                syncCapable = true;
-
-                // Query if the UITour API is working before we use the API
-                Mozilla.UITour.getConfiguration('sync', function (config) {
-
-                    // Variation #1: Firefox 31+ signed IN to Sync (default)
-                    if (config.setup) {
-                        swapState('state-fx-31-signed-in');
-                        state = 'Firefox 31 or Higher: Signed-In';
+    SyncPage.setBodyClass = function(stateClass) {
+        var classes = SyncPage.body.getAttribute('class');
```

Comment:
To add more clarity, 'class' is not a valid attribute. As per the spec, it's actually a property `className`. Modern browsers support it, but not every browser